app/controllers/membership_applications_controller.rb

```diff
@@ -45,11 +45,7 @@ def update
 
       new_upload_file params['uploaded_file'] if params['uploaded_file']
 
-      if changed_to_accepted?(params['membership_application'])
-        accept_application
-        redirect_to edit_membership_application_url(@membership_application)
-        return
-      end
+      @membership_application.applicant_updated_info! if (current_user.id == @membership_application.user.id)
```

Comment:
agree with both: shouldn't change the state if no changes were actually made, and we need to know if a user can edit their application (probably), and if a member can edit their application (don't know).